main.go

```diff
@@ -107,7 +111,7 @@ func main() {
 		setupLog.Error(err, "unable to create controller", "controller", "DNSRecord")
 		os.Exit(1)
 	}
-	dnsService := dns.NewService(mgr.GetClient(), dns.NewDefaultHostResolver(), defaultCtrlNS)
+	dnsService := dns.NewService(mgr.GetClient(), dns.NewSafeHostResolver(dns.NewDefaultHostResolver()), defaultCtrlNS)
```

Comment:
In case anyone wonders. The reason to wrap it with the SafeHostResolver is because it'll be share both by the Controller and the Webhook server so it avoids concurrency issues